pkg/resources/resources.go

```diff
@@ -0,0 +1,340 @@
+// Package resources implement the interface for accessing kubernetes resources
+package resources
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+
+	"github.com/grafana/xk6-kubernetes/pkg/utils"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/runtime/serializer/yaml"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/rest"
+)
+
+// UnstructuredOperations defines generic functions that operate on any kind of Kubernetes object
+type UnstructuredOperations interface {
+	Apply(manifest string) error
+	Create(obj map[string]interface{}) (map[string]interface{}, error)
+	Delete(kind string, name string, namespace string) error
+	Get(kind string, name string, namespace string) (map[string]interface{}, error)
+	List(kind string, namespace string) ([]map[string]interface{}, error)
+	Update(obj map[string]interface{}) (map[string]interface{}, error)
+}
+
+// StructuredOperations defines generic operations that handles runtime objects such as corev1.Pod.
+// It facilitates handling objects in the situations where their type is known as opposed to the
+// UnstructuredOperations
+type StructuredOperations interface {
```

Comment:
Should we not have an `apply` for structured as well?

Reply:
Apply reveives a yaml, not an object.  Please notice Structured operations are intended to be for internal use (of helpers) and are presently not exposed in the library.